utils/Azure/acrTools.ts

```diff
@@ -0,0 +1,277 @@
+ * @returns allRepos : an array of Repository objects that exist within the given registry
+ */
+export async function getAzureRepositories(registry: Registry): Promise<Repository[]> {
+    const allRepos: Repository[] = [];
+    let repo: Repository;
+    let azureAccount: AzureAccount = AzureCredentialsManager.getInstance().getAccount();
+    if (!azureAccount) {
+        return [];
+    }
+    const { accessToken, refreshToken } = await getTokens(registry);
+    if (accessToken && refreshToken) {
+
+        await request.get('https://' + registry.loginServer + '/v2/_catalog', {
+            auth: {
+                bearer: accessToken
+            }
+        }, (err, httpResponse, body) => {
+            if (body.length > 0) {
+                const repositories = JSON.parse(body).repositories;
+                for (let tempRepo of repositories) {
+                    repo = new Repository(registry, tempRepo, accessToken, refreshToken);
+                    allRepos.push(repo);
+                }
+            }
+        });
+    }
+    //Note these are ordered by default in alphabetical order
+    return allRepos;
+}
+
+/**
+ * @param registry : the registry to get credentials for
+ * @returns : the updated refresh and access tokens which can be used to generate a header for an API call
+ */
+export async function getTokens(registry: Registry): Promise<{ refreshToken: any, accessToken: any }> {
+    const subscription = getSub(registry);
+    const tenantId: string = subscription.tenantId;
+    let azureAccount: AzureAccount = AzureCredentialsManager.getInstance().getAccount();
+
+    const session: AzureSession = azureAccount.sessions.find((s, i, array) => s.tenantId.toLowerCase() === tenantId.toLowerCase());
+    const { accessToken, refreshToken } = await acquireToken(session);
+
+    //regenerates in case they have expired
+    if (accessToken && refreshToken) {
+        let refreshTokenARC;
+        let accessTokenARC;
+
+        await request.post('https://' + registry.loginServer + '/oauth2/exchange', {
+            form: {
+                grant_type: 'access_token_refresh_token',
+                service: registry.loginServer,
+                tenant: tenantId,
+                refresh_token: refreshToken,
+                access_token: accessToken
+            }
+        }, (err, httpResponse, body) => {
+            if (body.length > 0) {
+                refreshTokenARC = JSON.parse(body).refresh_token;
+            } else {
+                return;
+            }
+        });
+
+        await request.post('https://' + registry.loginServer + '/oauth2/token', {
+            form: {
+                grant_type: 'refresh_token',
+                service: registry.loginServer,
+                scope: 'registry:catalog:*',
+                refresh_token: refreshTokenARC
+            }
+        }, (err, httpResponse, body) => {
+            if (body.length > 0) {
+                accessTokenARC = JSON.parse(body).access_token;
```

Comment:
The existence of this and the following function with different names is rather confusing, can me use a single name and grant different parameters or differentiate them by how they acquire the token? Say acquireTokensWithLocalSession and acquireTokensWithRegistry?